libr/bin/p/bin_dex.c

```diff
@@ -1887,117 +1884,136 @@ static char *getname(RBinFile *bf, int type, int idx, bool sd) {
 	return NULL;
 }
 
-static RList *sections(RBinFile *bf) {
-	struct r_bin_dex_obj_t *bin = bf->o->bin_obj;
-	RList *ml = methods (bf);
-	RBinSection *ptr = NULL;
-	int ns, fsymsz = 0;
-	RList *ret = NULL;
+typedef struct {
+	ut64 addr;
+	ut64 size;
+} Section;
+
+static RBinSection *add_section(RList *ret, const char *name, Section s, int perm, char *format) {
+	r_return_val_if_fail (ret && name, NULL);
+	r_return_val_if_fail (s.addr < UT32_MAX, NULL);
+	r_return_val_if_fail (s.size > 0 && s.size < UT32_MAX, NULL);
+	RBinSection *ptr = R_NEW0 (RBinSection);
+	if (ptr) {
+		ptr->name = strdup (name);
+		ptr->paddr = ptr->vaddr = s.addr;
+		ptr->size = ptr->vsize = s.size;
+		ptr->perm = perm;
+		ptr->add = false;
+		if (format) {
+			ptr->format = format;
+		}
+		r_list_append (ret, ptr);
+	}
+	return ptr;
+}
+
+static void add_segment(RList *ret, const char *name, Section s, int perm) {
+	RBinSection *bs = add_section (ret, name, s, perm, NULL);
+	if (bs) {
+		bs->is_segment = true;
+		bs->add = true;
+	}
+}
+
+static bool dim(const char *name, Section *pre, Section *cur, Section *nex, Section *all) {
```

Comment:
what does this name `dim` mean?

Reply:
i took DIM from qbasic https://www.qb64.org/wiki/DIM it's kind of a cast, so it bounds the value in a variable to some restrictions of type and size. it's probably not the best name, any proposal?

Reply:
```suggestion
static bool is_valid_section(const char *name, Section *pre, Section *cur, Section *nex, Section *whole) {
```

Reply:
is not really just checkinf if its a valid section.its also trimming it's size and warning when something is wrong

Reply:
`validate_section` or `is_valid_section` or something like that :D `dim` is really hard name to understand. it remembers me about visual basic though :P

Reply:
And you don't like those memories? :P 

changed